src/hotspot/share/runtime/mutexLocker.cpp

```diff
@@ -273,22 +267,22 @@ void mutex_init() {
   def(PerfDataManager_lock         , PaddedMutex  , leaf,        true,  _safepoint_check_always); // used for synchronized access to PerfDataManager resources
 
   def(Threads_lock                 , PaddedMonitor, barrier,     true,  _safepoint_check_always);  // Used for safepoint protocol.
-  def(NonJavaThreadsList_lock      , PaddedMutex,   barrier,     true,  _safepoint_check_never);
+  def(NonJavaThreadsList_lock      , PaddedMutex,   leaf+1,      true,  _safepoint_check_never);
```

Comment:
Why do we need to change this rank? We now assert a lock should be _safepoint_check_always if the rank is >= nonleaf, but this is barrier so it should be good.

Reply:
This was leftover from a different change where I was trying to move all the _safepoint_check_never locks to leaf level.  In any case, I reverted this change and am retesting tier1-3.